### PR TITLE
Multiple not enrolled alert fixed

### PR DESCRIPTION
**Pull request by ahmedbilal**
Fix multiple enrolled alert.

**JIRA tickets**: 
* https://openedx.atlassian.net/browse/TNL-7249
* https://openedx.atlassian.net/browse/OSPR-4741

**Screenshots**: 
|  Before | After |
| --------- | ------- |
| image | image |

**Merge deadline**: "None"

**Testing instructions**:

1. Run [edx master devstack](https://github.com/edx/devstack).
2. Run [frontend-app-learning](https://github.com/edx/frontend-app-learning).
3. Visit http://localhost:18000 and login as staff.
4. Visit http://localhost:2000/course/course-v1:edX+DemoX+Demo_Course/outline.
5. Click on `Introduction` section and then `Demo Course Overview` link.
6. You should see only a single alert.
7. Press the back button of your browser.
8. Again perform `step 5`.
9. You should see only a single alert. If you see more than one, the issue is not fixed otherwise fixed.

**Reviewers**
- [ ] @giovannicimolin

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @ahmedbilal! I've created [OSPR-4741](https://openedx.atlassian.net/browse/OSPR-4741) to keep track of it in JIRA, where we prioritize reviews.  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).

Please let us know once your PR is ready for our review and all tests are green.

**Comment by natabene** [comment]
@ahmedbilal Thank you for your contribution. Please let me know once it is ready for edX review.

**Review by giovannicimolin — COMMENTED** [review]
@ahmedbilal I think this solution is a workaround the actual issue (message being added multiple times).

The main issue comes from the fact that useEnrollmentAlert and useLogistrationAlert are being called multiple times and add multiple messages to message store (react state on `UserMessagesProvider`).

I still think we should investigate a little more the root cause of the issue. Since this only happens when navigating back, I'm tempted to say that this is being caused by a missing cleanup of the state on re-renders somewhere.

I see that there's been a recent rework on these pages, so it might've been introduced there.
Can you try to take a more in-depth look at it?

@davidjoy @mikix Do you have any suspicions on what might be causing this issue?

**Comment by mikix** [comment]
I feel like I've seen it without a "back" action as well. But most reliable on a back.

File: src/user-messages/UserMessagesProvider.jsx [hunk 1]
```diff
@@ -1,6 +1,8 @@
 import React, { useState, useRef, useEffect } from 'react';
 import PropTypes from 'prop-types';
 
+import _ from 'lodash';
```
Comment (mikix) [inline src/user-messages/UserMessagesProvider.jsx]: I believe lodash and underscore are deprecated libraries for edX.

Comment (davidjoy) [reply src/user-messages/UserMessagesProvider.jsx]: We use lodash's individual packages.  i.e., we install `lodash.debounce` instead of the whole thing

**Comment by davidjoy** [comment]
Hmm, I'm finding that I can't reproduce this on master right now - @mikix, I'm getting unrelated errors when I go back and forth between course home and courseware which we may need to figure out/is a separate bug.  (It looks like we have two different levels of hydration for the 'course' model, which is causing courseware to barf when it can't find some data.)

I think some of the changes in this PR around making more granular alert topics make sense, and now that I'm reviewing the code on master, I feel like the places we're calling the `useEnrollmentAlert` aren't factored right, and that it should probably live on `OutlineTab.jsx` and `Course.jsx`.

I think I'd like to suggest we close this PR and https://openedx.atlassian.net/browse/TNL-7249 for now and instead fix some of our data issues/the factoring of where useEnrollmentAlert is called.  Then we can revisit if there are still issues.

**Comment by mikix** [comment]
I could believe that some of the refactoring has moved this bug around or left our alerts in non-ideal places. Sounds like a plan.

**Comment by ahmedbilal** [comment]
I am closing this PR as edX reviewer (@davidjoy, @mikix) suggested to close this PR as fixing data issues and factoring where useEnrollmentAlert is called would be better approach.

**Comment by openedx-webhooks** [comment]
@ahmedbilal Even though your pull request wasn’t merged, please take a moment to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/frontend-app-learning&entry.867055334=https://github.com/edx/frontend-app-learning/pull/90&entry.1484655318=https://github.com/ahmedbilal&entry.752974735=2020-06-22+10:19&entry.1917517419=2020-07-01+13:54&entry.2133058324=No) so we can improve your experience in the future.

**Pull request closed without merging**